cirq/circuits/circuit.py

```diff
@@ -666,16 +666,41 @@ def findall_operations_until_blocked(
             is_blocker: Callable[[ops.Operation], bool] = lambda op: False
     ) -> List[Tuple[int, ops.Operation]]:
         """
-        Finds all operations until a blocking operation is hit.  This returns
-        a list of all operations from the starting frontier until a blocking
-        operation is encountered.  An operation is part of the list if
-        it is involves a qubit in the start_frontier dictionary, comes after
-        the moment listed in that dictionary, and before any blocking
-        operations that involve that qubit.  Operations are only considered
-        to be blocking the qubits that they operate on, so a blocking operation
-        that does not operate on any qubit in the starting frontier is not
-        actually considered blocking.  See `reachable_frontier_from` for a more
-        in depth example of reachable states.
+        Finds all operations until a blocking operation is hit.
+
+        An operation is considered blocking if
+
+        a) It is in the 'light cone' of start_frontier.
+
+        AND
+
+        (
+
+            1) is_blocker returns a truthy value.
+
+            OR
+
+            2) It acts on a blocked qubit.
+        )
+
+        Every qubit acted on by a blocking operation is thereafter itself
+        blocked.
+
+
+        The notion of reachability here differs from that in
+        reachable_frontier_from in two respects:
+
+        1) An operation is not considered blocking only because it is in a
+            moment before the start_frontier of one of the qubits on which it
```

Comment:
This statement needs to be rephrased.  I've read it a few times now, and I am still confused as to what it means.  I would flip it around to make it a more direct statement.  Something like:

- An operation before the start_moment of one of the qubits is not considered blocking.

Reply:
That's not quite true. If an operation acts on two qubits in a moment that is before the start frontier of one but after the start frontier of the other, then it might be considered blocking.